data/provider/src/main/java/tech/pegasys/teku/api/blobselector/BlobSidecarSelectorFactory.java

```diff
@@ -96,7 +151,11 @@ public BlobSidecarSelector slotSelectorForAll(final UInt64 slot) {
             .getAllBlobSidecars(slot, indices)
             .thenApply(
                 blobSidecars ->
-                    blobSidecars.isEmpty() ? Optional.empty() : Optional.of(blobSidecars));
+                    blobSidecars.isEmpty()
+                        ? Optional.empty()
+                        : addMetaData(
+                            //TODO we don't care about metadata here...need to find a way to get around the lack of a block root?
```

Comment:
Yeah, we use slotSelectorForAll only in our internal `/teku/v1/beacon/blob_sidecars/{slot}` which doesn't have any of the metadata fields, so I would just add a comment why we don't care about the block root and just default it to ZERO as you do.